src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java

```diff
@@ -271,11 +271,11 @@ private static PkgImportControl load(InputSource source,
             return loader.getRoot();
         }
         catch (ParserConfigurationException | SAXException ex) {
-            throw new CheckstyleException("unable to parse " + uri
+            throw new CheckstyleException("unable to parse " + uri.toASCIIString()
                     + " - " + ex.getMessage(), ex);
```

Comment:
As a side note, this is ugly and just duplicates the exception message

Reply:
>  ex.getMessage()

good catch, yes, it would be good to remove this in this PR, but can be done separately.

Reply:
https://github.com/checkstyle/checkstyle/issues/13023